exonum/src/blockchain/mod.rs

```diff
@@ -381,6 +384,41 @@ impl Blockchain {
         }
         mount
     }
+
+    /// Saves raw message to the consensus messages cache
+    pub fn save_message(&mut self, raw: &RawMessage) {
+        let mut fork = self.fork();
+
+        {
+            let mut schema = Schema::new(&mut fork);
+            schema.consensus_messages_cache_mut().push(raw.clone());
+        }
+
+        // apply changes
+        self.merge(fork.into_patch()).unwrap();
```

Comment:
Perhaps it will be better to return `Result<(), Error>` instead of calling unwrap? In that case high level code can decide what to do with that error.

Reply:
Good point, thanks